test-utils/tests/tracing_macros.rs

```diff
@@ -0,0 +1,74 @@
+use std::{
+    thread::sleep,
+    time::{Duration, Instant},
+};
+
+use libcrux_test_utils::tracing::{EventType, MutexTrace, Trace as _, TraceEvent};
+
+mod str_labels {
+
+    use super::*;
+
+    lazy_static::lazy_static! {
+        static ref TRACE_STR: MutexTrace<&'static str, Instant> = Default::default();
+    }
+
+    #[libcrux_macros::trace_span("maybe_slow", TRACE_STR)]
+    fn maybe_slow_str() {
+        sleep(Duration::from_millis(100));
+    }
+
+    #[test]
+    fn test_tracing_str() {
+        maybe_slow_str();
+
+        let [open, close]: [TraceEvent<_, _>; 2] =
+            (*TRACE_STR).clone().report().try_into().unwrap();
+
+        assert_eq!(open.ty, EventType::SpanOpen);
+        assert_eq!(open.label, "maybe_slow");
+
+        assert_eq!(close.ty, EventType::SpanClose);
+        assert_eq!(close.label, "maybe_slow");
+
+        let run_time = close.at - open.at;
+        assert!(run_time.as_micros() > 99500);
+        assert!(run_time.as_micros() < 100500);
```

Comment:
```suggestion
        assert!(run_time.as_micros() > 99000);
        assert!(run_time.as_micros() < 101000);
```
Widening the acceptable interval a bit could help avoid sporadic failures

Reply:
I changed this to +- 0.7ms everywhere, I hope it's not an issue anymore.

Reply:
All the tests are passing now